src/psyclone/tests/dynamo0p3_stubgen_test.py

```diff
@@ -0,0 +1,800 @@
+    have any_w2 fields. In particular, check basis functions as these
+    have specific sizes associated with the particular function space'''
+    result = generate(os.path.join(BASE_PATH,
+                                   "testkern_multi_anyw2_basis_mod.f90"),
+                      api=TEST_API)
+    print(result)
+    expected_output = (
+        "      REAL(KIND=r_def), intent(in), dimension(3,ndf_any_w2,"
+        "np_xy,np_z) :: basis_any_w2\n"
+        "      REAL(KIND=r_def), intent(in), dimension(1,ndf_any_w2,"
+        "np_xy,np_z) :: diff_basis_any_w2")
+    assert expected_output in str(result)
+
+
+SIMPLE = (
+    "  MODULE simple_mod\n"
+    "    IMPLICIT NONE\n"
+    "    CONTAINS\n"
+    "    SUBROUTINE simple_code(nlayers, field_1_w1, ndf_w1, undf_w1,"
+    " map_w1)\n"
+    "      USE constants_mod, ONLY: r_def\n"
+    "      IMPLICIT NONE\n"
+    "      INTEGER, intent(in) :: nlayers\n"
+    "      INTEGER, intent(in) :: ndf_w1\n"
+    "      INTEGER, intent(in), dimension(ndf_w1) :: map_w1\n"
+    "      INTEGER, intent(in) :: undf_w1\n"
+    "      REAL(KIND=r_def), intent(out), dimension(undf_w1) ::"
+    " field_1_w1\n"
+    "    END SUBROUTINE simple_code\n"
+    "  END MODULE simple_mod")
+
+
+def test_stub_generate_working():
+    ''' check that the stub generate produces the expected output '''
+    result = generate(os.path.join(BASE_PATH, "simple.f90"),
+                      api=TEST_API)
+    print(SIMPLE)
+    print(result)
+    assert str(result).find(SIMPLE) != -1
+
+
+def test_stub_generate_working_noapi():
+    ''' check that the stub generate produces the expected output when
+    we use the default api (which should be dynamo0.3)'''
+    result = generate(os.path.join(BASE_PATH, "simple.f90"))
+    print(result)
+    assert str(result).find(SIMPLE) != -1
+
+
+SIMPLE_WITH_SCALARS = (
+    "  MODULE simple_with_scalars_mod\n"
+    "    IMPLICIT NONE\n"
+    "    CONTAINS\n"
+    "    SUBROUTINE simple_with_scalars_code(nlayers, rscalar_1, field_2_w1, "
+    "iscalar_3, ndf_w1, undf_w1, map_w1)\n"
+    "      USE constants_mod, ONLY: r_def\n"
+    "      IMPLICIT NONE\n"
+    "      INTEGER, intent(in) :: nlayers\n"
+    "      INTEGER, intent(in) :: ndf_w1\n"
+    "      INTEGER, intent(in), dimension(ndf_w1) :: map_w1\n"
+    "      INTEGER, intent(in) :: undf_w1\n"
+    "      REAL(KIND=r_def), intent(in) :: rscalar_1\n"
+    "      INTEGER, intent(in) :: iscalar_3\n"
+    "      REAL(KIND=r_def), intent(out), dimension(undf_w1) ::"
+    " field_2_w1\n"
+    "    END SUBROUTINE simple_with_scalars_code\n"
+    "  END MODULE simple_with_scalars_mod")
+
+
+def test_stub_generate_with_scalars():
+    ''' check that the stub generate produces the expected output when
+    the kernel has scalar arguments '''
+    result = generate(os.path.join(BASE_PATH, "simple_with_scalars.f90"),
+                      api=TEST_API)
+    print(result)
+    assert str(result).find(SIMPLE_WITH_SCALARS) != -1
```

Comment:
change to x in y  or x == y for this and all subsequent -1 tests in this file.

Reply:
Done.